### PR TITLE
Improve logic on discerning between long and wide format files

### PR DESCRIPTION
**Pull request by klevdiamanti**
***Note:** At this time we are unable to accept external contributions. For feature suggestions and any issues, please reach out to biostattools@olink.com.*

# Title: Improve logic on discerning between long and wide format files
**Problem:** In wide formatted files, when cell A2 was empty we would falsely identify file as long formatted, and misleading error would appear.

**Solution:** This pull request improves the robustness and error handling of the NPX file reading functions, particularly for edge cases where key cells (such as cell A2) are empty or contain unexpected values. It also updates and extends the test suite to cover these scenarios to reflect changes in test data or logic.

**Key Features:**

* Improved detection logic in `read_npx_format_get_format` to better handle cases where cell A2 is empty or NA, including updating the criteria for determining wide vs. long format and handling special cases for columns with "Version" or "Signature" in their names. [[1]](diffhunk://#diff-ce758d7eb7894c5844945e88628df9f2053914f8eb6a10cf7ba7375a0d73f46bR419-R452) [[2]](diffhunk://#diff-ce758d7eb7894c5844945e88628df9f2053914f8eb6a10cf7ba7375a0d73f46bL445-R461)
* Enhanced error handling in `read_npx_format_get_quant` to provide a clear error message when cell A2 is empty, guiding the user to set the `data_type` parameter if automatic detection fails.
* Added new tests for edge cases, such as when cell A2 is empty in wide-format files, ensuring that errors are thrown as expected and that format detection functions handle these cases gracefully. [[1]](diffhunk://#diff-b07cced680be4b87d7406a20eab3b035ad1427f95914c82557013d25be70e344R1445-R1506) [[2]](diffhunk://#diff-b07cced680be4b87d7406a20eab3b035ad1427f95914c82557013d25be70e344R1957-R2043) [[3]](diffhunk://#diff-b07cced680be4b87d7406a20eab3b035ad1427f95914c82557013d25be70e344R3470-R3507)
* Added and updated tests to check behavior when certain cells are set to NA, including modifications to synthetic test data to simulate these scenarios. [[1]](diffhunk://#diff-b07cced680be4b87d7406a20eab3b035ad1427f95914c82557013d25be70e344L2216-R2366) [[2]](diffhunk://#diff-b07cced680be4b87d7406a20eab3b035ad1427f95914c82557013d25be70e344L2337-R2488) [[3]](diffhunk://#diff-b07cced680be4b87d7406a20eab3b035ad1427f95914c82557013d25be70e344L2439-R2591)

## Checklist
- [X] Make sure you are requesting to **pull a feature/bugfix branch** (right side). This should not be main or develop.
- [X] Make sure you are make a pull request against either **main or develop** (left side). (Requesting to main should be reserved for bugfixes and new releases)
- [X] Add or update unit tests (if applicable)
- [X] Check your code with any unit tests (Run devtools::check() locally)
- [ ] Add neccessary documentation (if applicable)

## Type of changes

What type of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [X] Bugfix (non-breaking change which fixes an issue) (link the issue on the right)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code style update (formatting, renaming)
- [ ] Documentation Update 
- [ ] Other (explain)

## Further comments

We had to update some tests from the pathway enrichment as they were failing because of package updates:
* Adjusted expected output dimensions in pathway enrichment tests to reflect changes in the underlying test data or logic, ensuring tests remain accurate and up to date. [[1]](diffhunk://#diff-bc6ff0474e7cafee4533d7fc14744712709bf8eb89a7e36dec117ca40933d16eL423-R423) [[2]](diffhunk://#diff-bc6ff0474e7cafee4533d7fc14744712709bf8eb89a7e36dec117ca40933d16eL472-R472) [[3]](diffhunk://#diff-bc6ff0474e7cafee4533d7fc14744712709bf8eb89a7e36dec117ca40933d16eL575-R575)

### CODE REVIEW

File: OlinkAnalyze/R/read_npx_format.R [hunk 1]
```diff
@@ -773,17 +789,32 @@ read_npx_format_get_quant <- function(file,
...
 
+  if (all(is.na(data_cells))) {
+    cli::cli_abort(
+      message = c(
+        "x" = "Unable to recognize the quantification method from the input
+        file!",
+        "i" = "Cell 'A2' of the input file {.file {file}} in {.val {\"wide\"}}
+        format was likely empty. Expected one of:
+        {.val {get_olink_data_types(broad_platform = broad_platform)}}. Consider
+        setting {.arg data_type}."
```
Comment (kathy-nevola) [inline OlinkAnalyze/R/read_npx_format.R]: error still triggers when data_type is set, should work if we move it to after line 823
read_npx_format_get_quant works if A2 (data_cells) is "" or "Test" and data_type is set, but if A2 is NA or NULL and data_type is set, the function throws an error.

**Review by kathy-nevola — CHANGES_REQUESTED** [review]
Small change requested to support data if A2 is empty and data_type is provided with a valid argument

File: OlinkAnalyze/R/read_npx_format.R (hunk 1 above)
Comment (klevdiamanti) [reply OlinkAnalyze/R/read_npx_format.R]: Great catch. I've addressed it now with a more consistent solution.

**Comment by klevdiamanti** [comment]
Note that:

1. `R-CI-tests-pass` and `coverage-render-and-document` fail because of `rstatix` update fixed in PR #774 
2. `lintr` CI fails because of && expected in scripts `olink_anova.R`, `olink_lmer.R`, `Olink_one_non_parametric.R‎` and `olink_ordinal_regression.R`‎. These are fixed in #765

**Review by kathy-nevola — APPROVED** [review]
LGTM
